src/vmm/src/resources.rs

```diff
@@ -187,6 +189,23 @@ impl VmResources {
             return Err(VmConfigError::InvalidMemorySize);
         }
 
+        // The VM cannot have a memory size greater than the target size
+        // of the balloon device, if present.
+        if self.balloon.get().is_some() {
+            if machine_config
+                .mem_size_mib
+                .clone()
+                .unwrap_or(DEFAULT_MEM_SIZE_MIB)
+                < self
+                    .balloon
+                    .get_config()
+                    .map_err(|_| VmConfigError::IncompatibleBalloonSize)?
```

Comment:
It is a bit weird because it's a balloon related error in `VmConfigError`, so we have to be spot on with the name and description. What about `InvalidVmState`? If the `get_config()` fails at this stage, something is terribly wrong.

Reply:
I guess we should return the error that prevented us to return the config.